### ISSUE TITLE
SmallInteger>>isPrime is inefficient

### ISSUE THREAD

**Issue by johnhwoods** [guillep/pharo-core#11018]
**Bug description**
Smallinteger>>isPrime is inefficient.  The use of the well-known 6k±1 optimization is recomended

**To Reproduce**
Seconds to confirm the primality of the biggest SmallInteger prime using current method: 
`(Time microsecondsToRun: [ 1152921504606846883 isPrime ]) / 1.0e6.`

Seconds ditto using 6k±1 algorithm: I expect this to be approx half an order of magnitude faster.
```Smalltalk
(Time microsecondsToRun: [ | p s i  |
	p := 1152921504606846883.
	s := p sqrtFloor.
	i := 5.
	[ i < s ] whileTrue: [ 
		p \\ i = 0 ifTrue:[^false].
		i := i + 2.
		p \\ i = 0 ifTrue:[^false].
		i := i + 4.]]) / 1.0e6. 
```

**Expected development cost**
0, I will provide the new `isPrime` method and test it for both performance and accuracy against the 1000 largest SmallIntegers and a million random SmallIntegers.

**Comment by johnhwoods** [comment]
It's not really a 'bug' but I don't know how to apply a different label :-D

**Comment by johnhwoods** [comment]
Related: `Integer>>primeFactors` uses the same algorithm and can similarly be improved by using 6k±1

**Comment by Ducasse** [comment]
Improvements are always cool

**Comment by johnhwoods** [comment]
Done the code but just doing the tests before submitting PR.

**Comment by Ducasse** [comment]
Yes we love tests.... our life insurance :)

**Comment by johnhwoods** [comment]
With the 1000 biggest `SmallInteger`s, all `isPrime` results matched the old ones, total time saving ~45%
of 1 million random `SmallInteger` tests
100% match between result of `isPrime` 
87.9%  `isPrime` ran in 0µs to run in both old and new versions
0.035% `isPrime` ran in a few µs (310 in 1µs, 38 in 2µs, 3 in 3µs and 1 in 38µs)
6.0% ran faster in the new version at a 38.3% time saving.
6.0% ran slower, 258.1% time cost, but this is an artefact caused by the fact it only happens when the old test returns  0µs and the new one 1µs.  To confirm, the same numbers were run through the tests back to back and the new method was fractionally faster, but but both performed on average at about 0.5µs

So I'll commit, push and submit the PR.

**Comment by Ducasse** [comment]
Fixed by PR #11018

**Comment by johnhwoods** [comment]
yeah, apologies, obviously still got some of the old 'rona in the brain, there's a bug.  Have fixed it issue 11046 pr 11047 and submitted.
